frontend/src/pages/_organise/slug/edit/challenges/ChallengesForm.js

```diff
@@ -0,0 +1,394 @@
+import React, { useState, useCallback, useMemo } from 'react'
+
+import {
+    Paper,
+    Grid,
+    Box,
+    List,
+    ListItem,
+    ListItemText,
+    ListItemSecondaryAction,
+    IconButton,
+    Tooltip,
+    Typography,
+} from '@material-ui/core'
+import getSlug from 'speakingurl'
+import HighlightOffIcon from '@material-ui/icons/HighlightOff'
+import EditIcon from '@material-ui/icons/Edit'
+import SaveIcon from '@material-ui/icons/Save'
+import CloseIcon from '@material-ui/icons/Close'
+import TextInput from 'components/inputs/TextInput'
+import Button from 'components/generic/Button'
+import MarkdownInput from 'components/inputs/MarkdownInput'
+import ImageUpload from 'components/inputs/ImageUpload'
+import { SettingsBrightnessSharp } from '@material-ui/icons'
+
+export default ({ value, onChange }) => {
+    const [name, setName] = useState(undefined)
+    const [slug, setSlug] = useState(undefined)
+    const [partner, setPartner] = useState(undefined)
+    const [title, setTitle] = useState(undefined)
+    const [subtitle, setSubtitle] = useState(undefined)
+    const [description, setDescription] = useState(undefined)
+    const [insights, setInsights] = useState(undefined)
+    const [resources, setResources] = useState(undefined)
+    const [prizes, setPrizes] = useState(undefined)
+    const [criteria, setCriteria] = useState(undefined)
+    const [companyInfo, setCompanyInfo] = useState(undefined)
+    const [logo, setLogo] = useState(undefined)
```

Comment:
Näin isolle otukselle olisi ehkä jäkevämpää manageta yhtä statea sen sijaan, että deklaroidaan setterit kaikelle erikseen. Tulee ärsyttävän paljon riviä turhaa toistoa noissa editeissäkin, kun ei voi vaan spreadata yhtä state-objektia suoraan. Ei nyt ehkä tässä tapauksessa tarvitse muuttaa enää mun mielestä.